addenda/txp.go

```diff
@@ -0,0 +1,223 @@
+	AmountCents string
+	// AmountType is the tax information type ID for the tax amount
+	// Can be an integer or letter (e.g., "1", "2", "3", "S", "P", "I", etc.)
+	AmountType string
+}
+
+// TXP represents a Tax Payment addenda parsed from PaymentRelatedInformation
+// of an Addenda05 record. TXP is not a NACHA standard addenda type, but rather
+// a specific format for tax payment information within existing Addenda05 records.
+type TXP struct {
+	// TaxIdentificationNumber is the taxpayer's identification number
+	TaxIdentificationNumber string
+	// TaxPaymentTypeCode indicates the type of tax payment (e.g., FEDERAL, STATE, LOCAL)
+	TaxPaymentTypeCode string
+	// Date represents the tax period or payment date in YYYYMMDD format
+	Date string
+	// TaxInformationIDNumber is a unique identifier for the tax information
+	TaxInformationIDNumber string
+	// TaxAmounts is a slice of tax amounts with their types
+	TaxAmounts []TaxAmount
+	// TaxpayerVerification is the verification information
+	TaxpayerVerification string
+}
+
+// isNumeric checks if a string contains only numeric characters
+func isNumeric(s string) bool {
+	if s == "" {
+		return false
+	}
+	for _, r := range s {
+		if r < '0' || r > '9' {
+			return false
+		}
+	}
+	return true
+}
+
+// ErrInvalidTXPCharacter is returned when TXP contains invalid characters
+var ErrInvalidTXPCharacter = errors.New("invalid TXP character")
+
+// ErrInvalidTXPFormat is returned when the TXP format is invalid
+var ErrInvalidTXPFormat = errors.New("invalid TXP format")
+
+// TXPPrefix is the required prefix for TXP addenda records
+const TXPPrefix = "TXP*"
+
+// ParseTXP parses a TXP-formatted PaymentRelatedInformation string
+//
+// NACHA TXP Addenda Specification:
+// Data Element Reference Designator & Name    Content
+// 1  Segment Identifier                      'TXP'
+// 2  TXP01 Taxpayer Identification Number    XXXXXXXXX
+// 3  TXP02 Tax Payment Type Code             XXXXX
+// 4  TXP03 Tax Period End Date               YYMMDD
+// 5  TXP04 Amount Type (Tax Information ID)  XXXXX
+// 6  TXP05 Tax Amount                        $$$$$$$$cc
+// 7  TXP06 Amount Type (Tax Information ID)  XXXXX
+// 8  TXP07 Tax Amount                        $$$$$$$$cc
+// 9  TXP08 Amount Type (Tax Information ID)  XXXXX
+// 10 TXP09 Tax Amount                        $$$$$$$$cc
+// 11 TXP10 Taxpayer Verification             (Not used)
+//
+// Expected format: TXP*tax identification number*tax payment type code*date*type1*amount1*type2*amount2*type3*amount3*taxpayer verification\
+// Note: The total TXP addenda should be limited to 80 bytes
+func ParseTXP(paymentInfo string) (*TXP, error) {
+	if paymentInfo == "" {
+		return nil, ErrInvalidTXPFormat
+	}
+
+	// Must start with TXP prefix
+	if !strings.HasPrefix(paymentInfo, TXPPrefix) {
+		return nil, ErrInvalidTXPFormat
+	}
+
+	// Validate 80-byte limit for TXP addenda
+	if len(paymentInfo) > 80 {
+		return nil, ErrInvalidTXPFormat
+	}
```

Comment:
String lengths should always use [`utf8.RuneCountInString`](https://pkg.go.dev/unicode/utf8#RuneCountInString)